src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java

```diff
@@ -210,11 +214,14 @@ protected SamlUserAuthenticatedPrincipal createUserPrincipal(Object principal) {
         // note we expect default behavior, if provider has a converter this will break
         Saml2AuthenticatedPrincipal samlDetails = (Saml2AuthenticatedPrincipal) principal;
 
-        // upstream subject identifier
-        //        String subjectId = StringUtils.hasText(samlDetails.getFirstAttribute(SUBJECT_ATTRIBUTE))
-        //                ? samlDetails.getFirstAttribute(SUBJECT_ATTRIBUTE)
-        //                : samlDetails.getName();
-        String subjectId = samlDetails.getName();
+        // subjectId is optionally evaluated from an attribute defined in saml provider configurations
+        String subjectId = StringUtils.hasText(subAttributeName)
+            ? samlDetails.getFirstAttribute(subAttributeName)
+            : samlDetails.getName();
```

Comment:
imho it's better not to fallback to getName(): if the attribute is missing we should end with an error, otherwise we could wrongly identify the subject